app/controllers/application_controller.rb

```diff
@@ -929,6 +930,23 @@ def should_enforce_2fa?
       !current_user.has_any_second_factor_methods_enabled?
   end
 
+  def redirect_to_profile_if_required
+    return if request.format.json? && is_api?
+    return if !current_user
+    return if !current_user.needs_required_fields_check?
+
+    if current_user.populated_required_custom_fields?
+      current_user.bump_required_fields_version
+      return
+    end
+
+    redirect_path = path("/u/#{current_user.encoded_username}/preferences/profile")
+    if !request.fullpath.start_with?(redirect_path)
```

Comment:
I think we should address this in Part 3 when we bring 2FA and required fields focus modes together. (We currently have the exact same situation for 2FA requirement.) It's not really bricked since you can just input some dummy value to be free again.

Reply:
Yes but it's still a very bad experience for admins, and they may be reluctant to input dud data here. They could also have tons of fields to fill in. I would really prefer we address this in the current PR.

Reply:
Yes but it's still a very bad experience for admins, and they may be reluctant to input dud data here. They could also have tons of fields to fill in. I would really prefer we address this in the current PR.